[Bug] Oracle-CDC fails to obtain schema for the third table in Oracle 19c CDB/PDB multi-table CDC job

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

The issue is reproducible with a pure SeaTunnel config, without any external migration framework.
A single-table Oracle-CDC job works.
A two-table Oracle-CDC job works.
Three tables split into three independent Oracle-CDC sources work.
Three tables in one Oracle-CDC source fail consistently.
Using CDB root service as the JDBC URL, with debezium.database.pdb.name, fails earlier during catalog table discovery.

### SeaTunnel Version

Apache SeaTunnel: 3.0.0-SNAPSHOT
Java: OpenJDK 17
Source database: Oracle 19c CDB/PDB
CDB service: ORCL
PDB service: ORCLPDB
Source schemas:
LIB_A
LIB_B
Tables:
ORCLPDB.LIB_A.T_A1
ORCLPDB.LIB_A.T_A2
ORCLPDB.LIB_B.T_B1
Target sink: JDBC sink to GaussDB/OpenGauss-compatible database

### SeaTunnel Config

```conf
env {
  parallelism = 2
  job.mode = "STREAMING"
}

source {
  Oracle-CDC {
    plugin_output = "oracle_cdc_source"
    url = "jdbc:oracle:thin:@//<host>:1521/ORCLPDB"
    username = "<masked>"
    password = "<masked>"

    database-names = ["ORCLPDB"]
    schema-names = ["LIB_A", "LIB_B"]
    table-names = [
      "ORCLPDB.LIB_A.T_A1",
      "ORCLPDB.LIB_A.T_A2",
      "ORCLPDB.LIB_B.T_B1"
    ]

    snapshot.split.size = 2147483647
    use_select_count = true
    skip_analyze = true
    incremental.parallelism = 1

    debezium {
      database.url = "jdbc:oracle:thin:@//<host>:1521/ORCLPDB"
      database.dbname = "ORCLPDB"
      database.pdb.name = "ORCLPDB"
    }

    startup.mode = "initial"
  }
}

transform {
  TableRename {
    plugin_input = "oracle_cdc_source"
    plugin_output = "oracle_cdc_tables"
    convert_case = "LOWER"
  }

  FieldRename {
    plugin_input = "oracle_cdc_tables"
    plugin_output = "oracle_cdc_fields"
    convert_case = "LOWER"
  }
}

sink {
  Jdbc {
    plugin_input = "oracle_cdc_fields"
    url = "jdbc:gaussdb://<host>:8000/postgres"
    driver = "com.huawei.gaussdb.jdbc.Driver"
    compatible_mode = "OpenGauss"
    dialect = "OpenGauss"
    database = "postgres"
    table = "public.${table_name}"
    username = "<masked>"
    password = "<masked>"
    batch_size = 1000
    field_ide = "LOWERCASE"
    generate_sink_sql = true
    enable_upsert = true
    primary_keys = ["id"]
  }
}
```

### Running Command

```shell
bash bin/seatunnel.sh --config 19c_full_incr.config -m local
```

### Error Exception

```log
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can't obtain schema for table ORCLPDB.LIB_B.T_B1
    at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.utils.OracleSchema.readTableSchema(OracleSchema.java:94)
    at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.utils.OracleSchema.getTableSchema(OracleSchema.java:59)
    at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleDialect.queryTableSchema(OracleDialect.java:107)
    at org.apache.seatunnel.connectors.cdc.base.source.reader.external.JdbcSourceFetchTaskContext.registerDatabaseHistory(JdbcSourceFetchTaskContext.java:174)
    at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.reader.fetch.OracleSourceFetchTaskContext.configure(OracleSourceFetchTaskContext.java:99)
    at org.apache.seatunnel.connectors.cdc.base.source.reader.external.IncrementalSourceScanFetcher.submitTask(IncrementalSourceScanFetcher.java:85)
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-oracle/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/utils/OracleSchema.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-oracle/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/utils/OracleSchema.java
@@ -74,14 +74,15 @@ public class OracleSchema {
                     null,
                     false);
             for (TableId id : tables.tableIds()) {
-                if (tableMap.containsKey(id)) {
+                TableId tableMapId = resolveTableId(id, tableId, tableMap);
+                if (tableMap.containsKey(tableMapId)) {
                     Table table =
                             CatalogTableUtils.mergeCatalogTableConfig(
-                                    tables.forTable(id), tableMap.get(id));
+                                    tables.forTable(id), tableMap.get(tableMapId));
                     TableChanges.TableChange tableChange =
                             new TableChanges.TableChange(
                                     TableChanges.TableChangeType.CREATE, table);
-                    schemasByTableId.put(id, tableChange);
+                    schemasByTableId.put(tableMapId, tableChange);
                 }
             }
         } catch (SQLException e) {
@@ -95,5 +96,20 @@ public class OracleSchema {
         }
 
         return schemasByTableId.get(tableId);
+    }
+
+    static TableId resolveTableId(
+            TableId readTableId, TableId requestedTableId, Map<TableId, ?> tableMap) {
+        if (tableMap.containsKey(readTableId)) {
+            return readTableId;
+        }
+
+        TableId readTableIdWithRequestedCatalog =
+                new TableId(requestedTableId.catalog(), readTableId.schema(), readTableId.table());
+        if (tableMap.containsKey(readTableIdWithRequestedCatalog)) {
+            return readTableIdWithRequestedCatalog;
+        }
+
+        return readTableId;
     }
 }

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-oracle/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/utils/OracleUtilsTest.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-oracle/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/utils/OracleUtilsTest.java
@@ -26,6 +26,9 @@ import org.junit.jupiter.api.Test;
 
 import io.debezium.relational.TableId;
 
+import java.util.Collections;
+import java.util.Map;
+
 public class OracleUtilsTest {
     @Test
     public void testSplitScanQuery() {
@@ -69,5 +72,16 @@ public class OracleUtilsTest {
                         true);
         Assertions.assertEquals(
                 "SELECT * FROM \"schema1\".\"table1\" WHERE \"id\" >= ?", splitScanSQL);
+    }
+
+    @Test
+    public void testResolveTableIdWithRequestedCatalog() {
+        TableId requestedTableId = TableId.parse("ORCLPDB.LIB_B.T_B1");
+        TableId readTableIdWithoutCatalog = new TableId(null, "LIB_B", "T_B1");
+        Map<TableId, ?> tableMap = Collections.singletonMap(requestedTableId, null);
+
+        Assertions.assertEquals(
+                requestedTableId,
+                OracleSchema.resolveTableId(readTableIdWithoutCatalog, requestedTableId, tableMap));
     }
 }

